### PR TITLE
hw-mgmt: scripts: Introduce support of SN4280 platform

### PR DESCRIPTION
**Pull request by ciju-nvidia**
Provide platform support for Nvidia Smart Switch SN4280.

The Smart Switch equipped with:
- Nvidia COME module based on AMD EPYC™ Embedded 3451 CPU.
- Nvidia Spectrum-3 ASIC.
- Four DPUs, each equipped with Nvidia BF3 ARM based processor and with Lattice LFD2NX-40 FPGA device.
- 28xQSFP-DD external ports.
- Two power supplies.
- Four cooling drawers.

### CODE REVIEW

**Comment by ciju-nvidia** [comment]
RERUN

File: usr/lib/udev/rules.d/50-hw-management-events.rules [hunk 1]
```diff
@@ -610,9 +610,15 @@ SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*
 SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*/*-0068/hwmon/hwmon*", ACTION=="add", RUN+="/usr/bin/hw-management-chassis-events.sh add voltmon4 %S %p"
 SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*/*-0068/hwmon/hwmon*", ACTION=="remove", RUN+="/usr/bin/hw-management-chassis-events.sh rm voltmon4 %S %p"
 
+SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*/*-0069/hwmon/hwmon*", ACTION=="add", RUN+="/usr/bin/hw-management-chassis-events.sh add voltmon1 %S %p dpu"
+SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*/*-0069/hwmon/hwmon*", ACTION=="remove", RUN+="/usr/bin/hw-management-chassis-events.sh rm voltmon1 %S %p dpu"
```
Comment (sholeksandr) [inline usr/lib/udev/rules.d/50-hw-management-events.rules:614]: I think we should stop add event handlers for each new sensor address. We have devtree support which should resolve all sesor address/name.

File: usr/lib/udev/rules.d/50-hw-management-events.rules [hunk 2]
```diff
@@ -610,9 +610,15 @@ SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*
...
+
 SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*/*-006a/hwmon/hwmon*", ACTION=="add", RUN+="/usr/bin/hw-management-chassis-events.sh add voltmon5 %S %p"
 SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*/*-006a/hwmon/hwmon*", ACTION=="remove", RUN+="/usr/bin/hw-management-chassis-events.sh rm voltmon5 %S %p"
 
+SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*/*-006a/hwmon/hwmon*", ACTION=="add", RUN+="/usr/bin/hw-management-chassis-events.sh add voltmon2 %S %p dpu"
+SUBSYSTEM=="hwmon", DEVPATH=="/devices/platform/mlxplat/i2c_mlxcpld*/i2c-*/i2c-*/*-006a/hwmon/hwmon*", ACTION=="remove", RUN+="/usr/bin/hw-management-chassis-events.sh rm voltmon2 %S %p dpu"
```
Comment (sholeksandr) [inline usr/lib/udev/rules.d/50-hw-management-events.rules:620]: Conflict with 2 previous rules:  use same filters SUBSYSTEM, DEVPATH, ACTION

Comment (ciju-nvidia) [reply usr/lib/udev/rules.d/50-hw-management-events.rules:620]: The udev rules have many issues and there are multiple events for the same sensor. This need to be rectified moving forward.
For the time being, I've added a parameter 'dpu' at end to differentiate that the sensors belong to DPU.

File: recipes-kernel/linux/kconfig_6_1.txt [hunk 3]
```diff
@@ -28,6 +28,7 @@ CONFIG_MLXREG_HOTPLUG=m
 CONFIG_MLXREG_IO=m
 CONFIG_MLX_WDT=m
 CONFIG_MLXREG_LC=m
+CONFIG_MLXREG_DPU=m
```
[the remark this answers is not in the document]
Comment (ciju-nvidia) [reply recipes-kernel/linux/kconfig_6_1.txt:31]: I'm not sure.

I think, we will move it to Downstream if required.

File: recipes-kernel/linux/kconfig_5_10.txt [hunk 4]
```diff
@@ -27,6 +27,7 @@ CONFIG_MLX_PLATFORM=m
 CONFIG_MLXREG_HOTPLUG=m
 CONFIG_MLXREG_IO=m
 CONFIG_MLX_WDT=m
+CONFIG_MLXREG_DPU=m
```
[the remark this answers is not in the document]
Comment (ciju-nvidia) [reply recipes-kernel/linux/kconfig_5_10.txt:30]: I'm not sure. I think, we will move it to Downstream if required.

**Review by sholeksandr — APPROVED** [review]

**Pull request merged**